### PR TITLE
Add spec URLs for web/accessibility subtree

### PR DESCRIPTION
**Pull request by sideshowbarker**
Part of https://github.com/mdn/content/issues/13126

### CODE REVIEW

File: files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ tags:
   - ARIA property
   - aria-activedescendant
   - NeedsExample
+spec-urls: >-
+  https://w3c.github.io/aria/#aria-activedescendant,
+  https://w3c.github.io/aria-practices/#kbd_focus_activedescendant
```
Comment (teoli2003) [inline files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:13]: The second spec leads to _unknown specification_. Is there a PR in w3c/browser-spec to fix it?

**Review by teoli2003 — APPROVED** [review]
Approving, but I'm waiting for the answer to my question to merge it.

File: files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md (hunk 1 above)
Comment (sideshowbarker) [reply files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:13]: > The second spec leads to _unknown specification_. Is there a PR in w3c/browser-spec to fix it?

No, there’s yet. So yeah, that’s one wrinkle with landing the changes for this particular subtree — the WAI-ARIA Authoring Practices doc isn’t yet in https://github.com/w3c/browser-specs, so our Specifications table entries for it  are going to show up as _Unknown specification_ until we get it into https://github.com/w3c/browser-specs. cc @Elchi3

File: files/en-us/web/accessibility/aria/attributes/aria-multiline/index.md [hunk 2]
```diff
@@ -8,13 +8,14 @@ tags:
...
   - Reference
+spec-urls: https://w3c.github.io/aria/#aria-multiline
 ---
 
 The `aria-multiline` attribute indicates whether a `textbox` accepts multiple lines of input or only a single line.
 
 ## Description
 
-The default behavior of the <kbd>Enter</kbd> or <kbd>Return</kbd> key differs between single-line and multi-line text fields. When user focus is in a single-line `{{htmlelement("input/text", '&lt;input type="text">')}}`, an <kbd>Enter</kbd> or <kbd>Return</kbd> keystroke usually submits the form.
+The default behavior of the <kbd>Enter</kbd> or <kbd>Return</kbd> key differs between single-line and multi-line text fields. When user focus is in a single-line `{{htmlelement("input/text", '&lt;input type="text"&gt;')}}`, an <kbd>Enter</kbd> or <kbd>Return</kbd> keystroke usually submits the form.
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/attributes/aria-multiline/index.md:18]: In a few places, this `>`→`&gt;` change crept in, past the process I came up with to filter out unrelated changes. It seems innocuous enough — but if we prefer, I could revert those instances out of this PR.

File: files/en-us/web/accessibility/aria/roles/dialog_role/index.md [hunk 3]
```diff
@@ -119,18 +122,15 @@ When the dialog is correctly labeled and focus is moved to an element (often an
...
+{{Specifications}}
 
 Coming soon
 
 ## See also
 
 - [ARIA: alertdialog role](/en-US/docs/web/Accessibility/ARIA/roles/alertdialog_role)
 - [Using the alertdialog role](/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role)
-- {{HTMLElement('dialog', 'The HTML <code>&lt;dialog></code> element')}}
+- {{HTMLElement('dialog', 'The HTML <code>&lt;dialog&gt;</code> element')}}
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/dialog_role/index.md:133]: Another `>`→`&gt;` instance

File: files/en-us/web/accessibility/aria/roles/grid_role/index.md [hunk 4]
```diff
@@ -605,7 +603,7 @@ Even if the keyboard use is properly implemented, some users might not be aware
 - [ARIA: `gridcell` role](/en-US/docs/Web/Accessibility/ARIA/Roles/gridcell_role)
 - [ARIA: `rowheader` role](/en-US/docs/Web/Accessibility/ARIA/Roles/rowheader_role)
 - [ARIA: columnheader role](/en-US/docs/Web/Accessibility/ARIA/Roles/columnheader_role)
-- {{HTMLElement('table','HTML <code>&lg;table></code> element')}}
+- {{HTMLElement('table','HTML <code>≶table&gt;</code> element')}}
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/grid_role/index.md:606]: `>`→`&gt;`

File: files/en-us/web/accessibility/aria/roles/row_role/index.md [hunk 5]
```diff
@@ -69,9 +70,9 @@ To create an interactive widget that has a tabular structure, use the grid patte
 - [role="gridcell"](/en-US/docs/Web/Accessibility/ARIA/Roles/gridcell_role)
   - : A cell in a row within a grid or treegrid.
 - [role="columnheader"](/en-US/docs/Web/Accessibility/ARIA/Roles/columnheader_role)
-  - : A header cell that is the structural equivalent of the HTML {{HTMLElement('th')}} element with a column scope ({{HTMLElement('tr', '<code>&lt;tr  scope="col"></code>')}}). Unlike a plain cell, the columnheader role establishes a relationship between it and all cells in the corresponding column.
+  - : A header cell that is the structural equivalent of the HTML {{HTMLElement('th')}} element with a column scope ({{HTMLElement('tr', '<code>&lt;tr  scope="col"&gt;</code>')}}). Unlike a plain cell, the columnheader role establishes a relationship between it and all cells in the corresponding column.
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/row_role/index.md:73]: `>`→`&gt;`

File: files/en-us/web/accessibility/aria/roles/row_role/index.md [hunk 6]
```diff
@@ -69,9 +70,9 @@ To create an interactive widget that has a tabular structure, use the grid patte
...
 - [role="rowheader"](/en-US/docs/Web/Accessibility/ARIA/Roles/rowheader_role)
-  - : A header cell that is the structural equivalent of the HTML {{HTMLElement('th')}} element with a row scope ({{HTMLElement('tr', '<code>&lt;tr  scope="row"></code>')}}). Unlike a plain cell, the rowheader role establishes a relationship between it and all cells in the corresponding row.
+  - : A header cell that is the structural equivalent of the HTML {{HTMLElement('th')}} element with a row scope ({{HTMLElement('tr', '<code>&lt;tr  scope="row"&gt;</code>')}}). Unlike a plain cell, the rowheader role establishes a relationship between it and all cells in the corresponding row.
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/row_role/index.md:75]: `>`→`&gt;`

File: files/en-us/web/accessibility/aria/roles/row_role/index.md [hunk 7]
```diff
@@ -195,14 +196,11 @@ none
...
-| Specification                                                                        | Status                                           |
-| ------------------------------------------------------------------------------------ | ------------------------------------------------ |
-| {{SpecName("ARIA","#row","ARIA row role")}}                             | {{Spec2('ARIA')}}                         |
-| {{SpecName("ARIA Authoring Practices 1.2","#row","ARIA row role")}} | {{Spec2('ARIA Authoring Practices 1.2')}} |
+{{Specifications}}
 
 ## See also
 
-- The HTML {{HTMLElement('tr', '<code>&lt;tr></code> table row')}} element
+- The HTML {{HTMLElement('tr', '<code>&lt;tr&gt;</code> table row')}} element
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/row_role/index.md:203]: `>`→`&gt;`

File: files/en-us/web/accessibility/aria/roles/sectionhead_role/index.md [hunk 8]
```diff
@@ -17,17 +18,15 @@ The **`sectionhead` role**, an abstract role, is superclass role for labels or s
...
+The structural `sectionhead` role is an abstract role for the subclass roles that identify the labels or summaries of the sections they label. The role must not be used. The four subclasses -- [`columnheader`](/en-US/docs/Web/Accessibility/ARIA/Roles/columnheader_role), [`heading`](/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role), [`rowheader`](/en-US/docs/Web/Accessibility/ARIA/Roles/rowheader_role), and [`tab`](/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role). There are HTML element equivalents, like, {{HTMLElement('th', '<code>&lt;th scope="col"&gt;</code>')}} for `columnheader`, {{HTMLElement('th', '<code>&lt;th scope="row"&gt;</code>')}} for rowheader, and any of the HTML headings, {{HTMLElement('h1')}} thru {{HTMLElement('h6')}} for `heading`. The `tab` role does not currently have an HTML equivalent.
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/sectionhead_role/index.md:21]: `>`→`&gt;`

File: files/en-us/web/accessibility/aria/roles/structure_role/index.md [hunk 9]
```diff
@@ -43,20 +42,12 @@ Do not use `role="structure"`. Do use HTML and subclass structure roles.
...
-1. [**WAI-ARIA roles**](/en-US/docs/Web/Accessibility/ARIA/Roles)
-
-    {{ListSubpagesForSidebar("/en-US/docs/Web/Accessibility/ARIA/Roles")}}
+<div class="hidden"><!-- these shouldn't be used so we shouldn't link to them 
+    - [ARIA: `application` role](/en-US/docs/Web/Accessibility/ARIA/Roles/application_role)
+    - [ARIA: `document` role](/en-US/docs/Web/Accessibility/ARIA/
+    "Roles/document_role)
+    - [ARIA: `rowgroup` role](/en-US/docs/Web/Accessibility/ARIA/Roles/rowgroup_role)
+    - [ARIA: `separator` role](/en-US/docs/Web/Accessibility/ARIA/Roles/separator_role)
+    -->]</div>
```
Comment (sideshowbarker) [inline files/en-us/web/accessibility/aria/roles/structure_role/index.md:52]: This looks like possible breakage. Regardless, I can revert it from the PR.

File: files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md (hunk 1 above)
Comment (Elchi3) [reply files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:13]: Do you want to propose to add it to https://github.com/w3c/browser-specs? Or does it not qualify there?

Comment (sideshowbarker) [reply files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:13]: >> ```  
>> https://w3c.github.io/aria-practices/#kbd_focus_activedescendant
>> ```
>
> Do you want to propose to add it to [w3c/browser-specs](https://github.com/w3c/browser-specs)? Or does it not qualify there?

I think it doesn’t qualify, and we should not propose to add it to browser-specs — given that the stated scope there is:

> technical Web specifications that are directly implemented or that will be implemented by Web browsers

There are yet no other guide-type documents like that in browser-specs, as far as I can see.

We have a related issue with a number of RFCs referenced in the `web/http` docs; see https://github.com/mdn/content/issues/13126#issuecomment-1048498884.

And relevant to that, https://github.com/w3c/browser-specs#spec-selection-criteria explicitly says:

> most IETF specs are likely out of scope

So, I guess that means we’re gonna need to add some special-casing in our own Yari code for these cases.

Given that we already have `SpecData.json`, maybe the way we could handle these cases is:

1. We trim `SpecData.json` down to just the specs that are not also in browser-specs.
2. We add another condition in the https://github.com/mdn/yari/blob/738c96143825ec3e034740002728ace0f6017248/build/document-extractor.js#L488-L497 code that looks up the spec URL in `SpecData.json` to get the spec name. The spec name is the one single thing we need, and the lookup is dead simple: one line of code to both load the JSON from `SpecData.json` and to do the lookup. And that code doesn’t ever even need in the vast majority of cases — but instead only in the rare cases when an article is citing a spec that’s not in browser-specs.

…and now that I’ve said all that, I realize it’s taken me longer to write this comment than it would have to just write the actual code for doing it. So anyway at this point I’ll write up a patch for the `build/document-extractor.js` that does what I described above. Then we have something concrete to look at and test with.

Comment (teoli2003) [reply files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:13]: > Given that we already have `SpecData.json`, maybe the way we [should use it]
Or we can create an `AdditionalSpecs.json` with the same structure as browsers-spec and load both of them in the macro. (And ultimately to get rid of `SpecData.json` at the same time as the eponymous macro).

That way we would have fewer special cases in macros (only the loading is special) and we could reuse any tool using w3c/browser-specs.

Disadvantages are the structure of w3c/browser-specs may be overkill here, and of course, we keep SpecData.json in parallel for some time.

I have no strong feelings here.

Comment (sideshowbarker) [reply files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:13]: For now I went ahead and opened https://github.com/mdn/yari/pull/5314 with a patch that does the `SpecData.json` lookup I described in my earlier comment.

That has the advantage of being a code-only change — it works with our existing data, without needing anything more.

> Or we can create an `AdditionalSpecs.json` with the same structure as browsers-spec and load both of them in the macro.

That doesn’t sound like a bad idea in principle — but I guess wouldn’t want to block this set of content PRs on first getting resolution about if we want to do that. It seems like if we use the `SpecData.json` way for now, it won’t be doing any harm, and it doesn’t prevent us from later deciding to create a browser-specs-compatible `AdditionalSpecs.json` if we end up deciding we want to.

> (And ultimately to get rid of `SpecData.json` at the same time as the eponymous macro).

Yeah, though we would be able to get rid of the macros without needing to get rid of `SpecData.json` too.

> That way we would have fewer special cases in macros (only the loading is special) and we could reuse any tool using w3c/browser-specs.

Being able to reuse any tool using browser-specs sounds like a great idea in principle — and I can imagine we might eventually have some specific need for that characteristic — but for now, I’m not aware that we have any specific needs to make our “additional specs” data store browser-specs-compatible. If we were designing it from scratch, making it browser-specs-compatible would seem to clearly be the right choice. But I don’t know how much switching away from the `SpecData.json` data store we already have would buy us right now.

> Disadvantages are the structure of w3c/browser-specs may be overkill here, and of course, we keep SpecData.json in parallel for some time.

Yeah — when it comes down to it, right now even `SpecData.json` has more than we need, and is a bit of overkill. What I mean is, we no longer have need for the `status` field in `SpecData.json`, nor do we even have need for the he abbreviated names that are the top-level keys — because when we quit using the `SpecName` macro, we won’t need those abbreviated names any longer. So for our current needs, we could actually get by with an even-simpler object structure where the keys are the spec URLs and the values are the spec names:

```json
{
  "https://w3c.github.io/aria-practices/": "WAI-ARIA Authoring Practices",
}
```
So unless I’m missing something and we do have use for something more than that, then yeah, replicating the structure of browser-specs would seem to be overkill for us, as far as our actual current needs.

**Comment by sideshowbarker** [comment]
This is now just blocked on https://github.com/mdn/yari/pull/5314

**Comment by estelle** [comment]
the conflicted pages are complete rewrites of the page. merging this will create a conflict in almost all these PRs: https://github.com/mdn/content/pulls?q=is%3Apr+is%3Aopen+label%3AContent%3AAccessibility

**Comment by sideshowbarker** [comment]
> the conflicted pages are complete rewrites of the page. merging this will create a conflict in almost all these PRs: [https://github.com/mdn/content/pulls?q=is%3Apr+is%3Aopen+label%3AContent%3AAccessibility](https://github.com/mdn/content/pulls?q=is%3Apr+is%3Aopen+label%3AContent%3AAccessibility+sort%3Aupdated-desc)

OK, I can rebase it and resolve the merge conflicts

**Comment by sideshowbarker** [comment]
@estelle This one’s now ready for re-review

File: files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md [hunk 10]
```diff
@@ -6,8 +6,12 @@ tags:
   - ARIA
   - ARIA attribute
   - ARIA property
+  - Reference
```
Comment (queengooborg) [inline files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md:9]: The tag changes look good but we should probably defer these to another PR -- could you undo these changes?

Comment (sideshowbarker) [reply files/en-us/web/accessibility/aria/attributes/aria-activedescendant/index.md]: > The tag changes look good but we should probably defer these to another PR -- could you undo these changes?

Yup, thanks — removed all those from this PR branch

**Review by queengooborg — APPROVED** [review]
LGTM, thank you!
